src/Elastic.Apm.EntityFrameworkCore/EfCoreDiagnosticListener.cs

```diff
@@ -12,6 +13,11 @@ namespace Elastic.Apm.EntityFrameworkCore
 	internal class EfCoreDiagnosticListener : IDiagnosticListener
 	{
 		private readonly ConcurrentDictionary<Guid, ISpan> _spans = new ConcurrentDictionary<Guid, ISpan>();
+
+		public EfCoreDiagnosticListener(AbstractLogger logger) => Logger = logger;

```

Comment:
Another one: Similarly to `ElasticCoreListeners.cs` I think we should make this optional and default to `Agent.Config.Logger` (or `Agent.Config` if we change it...).